ussologin/ussologin.go

```diff
@@ -69,13 +71,16 @@ var loginForm = form.Form{
 		otpKey: environschema.Attr{
 			Description: "Two-factor auth",
 			Type:        environschema.Tstring,
+			Mandatory:   true,
 			Group:       "2",
 		},
 	},
 }
 
-// doSignedRequest performs a GET request to the given URL signed using the provided token.
-func doSignedRequest(client *http.Client, ussoAuthUrl string, tok *usso.SSOData, u *url.URL) error {
+// LoginWithToken completes a login attempt using tok. The ussoAuthURL
+// should have been obtained from the target using a LoginMethods
```

Comment:
// obtained from the UbuntuSSOOAuth field in a response
// to a LoginMethods request from the target service.

?